roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -10,3 +10,24 @@
       openshift_hosted_registry_storage_s3_bucket and
       openshift_hosted_registry_storage_s3_region are required
   when: openshift.hosted.registry.storage.s3.bucket | default(none) is none or openshift.hosted.registry.storage.s3.region | default(none) is none
+
+# If cloudfront is being used, fail if we don't have all the required variables
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_cloudfront_baseurl and
+      openshift_hosted_registry_storage_s3_cloudfront_keypairid and
+      openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile are required
+  when:
+    - openshift_hosted_registry_storage_s3_cloudfront_baseurl is defined
+    - openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile | default(none) is none or openshift_hosted_registry_storage_s3_cloudfront_keypairid | default(none) is none
```

Comment:
Instead of fail, it might be better to use assert here: http://docs.ansible.com/ansible/assert_module.html